server/docs/getting-started.md

```diff
@@ -585,37 +444,132 @@ async function run() {
 
 ```
 
-Finally, we can include a game reset:
+Now we're ready to describe the `initialState` and call the `proposeVirtualInstall()` method:
+
 
 ```typescript
-let web3Provider, nodeProvider, account, currentGame;
+async function install() {
+  resetGameState();
+
+  let cfProvider = new cf.Provider(nodeProvider);
+  let appFactory = new cf.AppFactory(contractAddress, {
+    actionEncoding: " ",
+    stateEncoding: " "
+  }, cfProvider);
+
+  proposeInstall(appFactory);
+}
 
 ...
 
-function resetGameState() {
-  currentGame = {
-    highRollerState: {
-      stage: HighRollerStage.PRE_GAME
+const initialState = {
+      playerAddrs: [
+        deriveAddress(
+          account.nodeAddress
+        ),
+        deriveAddress(
+          nodeAddress
+        )
+      ],
+      stage: HighRollerStage.PRE_GAME,
+      salt: HashZero,
+      commitHash: HashZero,
+      playerFirstNumber: 0,
+      playerSecondNumber: 0
+    };
+
+async function proposeInstall(appFactory) {
+  const { intermediary, nodeAddress } = await getOpponentData();
+  const depositAmount = '0.00001';
+
+  await appFactory.proposeInstallVirtual({
+    initialState,
+    proposedToIdentifier: nodeAddress,
+    asset: {
+      assetType: 0 /* AssetType.ETH */
     },
-    salt: generateSalt()
-  };
+    peerDeposit: parseEther(depositAmount),
+    myDeposit: parseEther(depositAmount),
+    timeout: 172800,
+    intermediaries: [intermediary]
+  });
 }
 ```
 
-where the `generateSalt()` function looks like
 
+### The Provider's on() Method
+
+The `install()` function is also where we instruct the cfProvider to **listen** in the channel for
+
+- successful `proposeInstallVirtual()`
+- changes in state in the channel
+
+and to react to those changes via the `Provider` method `on()`. 
+
+
+```typescript
+    async function install() {
+      resetGameState();
+
+      let cfProvider = new cf.Provider(nodeProvider);
+      let appFactory = new cf.AppFactory(contractAddress, {
+        actionEncoding: " ",
+        stateEncoding: " "
+      }, cfProvider);
+
+      proposeInstall(appFactory);
+
+      cfProvider.on('installVirtual', onInstallEvent);
+      cfProvider.on('updateState', onUpdateEvent);
+    }
 ```
-function generateSalt() {
-  return ethers.utils.bigNumberify(ethers.utils.randomBytes(32)).toHexString();
+
+When cfProvider detects `installVirtual` (when the bot accepts our `proposeInstallVirtual()`) it calls the function `onInstallEvent()`; when it detects `updateState` (updates in state in the virtual channel), it calls `onUpdateEvent()`.
```

Comment:
These parentheses make the sentence hard to read; I would try to restructure without parentheses

Reply:
Also again the `()` confuses the argments to the function